backend/app.js

```diff
@@ -11,19 +11,31 @@ const urlencodedParser = bodyParser.urlencoded({ extended: false })
 const app = express()
 const port = 3000
 
-require('./bot');
-
 const addCrossDomainHeaders = function(req, res, next) {
-    res.header('Access-Control-Allow-Origin', "*")
+    res.header('Access-Control-Allow-Origin', '*')
     res.header('Access-Control-Allow-Methods', 'GET,PUT,POST,DELETE')
     res.header('Access-Control-Allow-Headers', 'Content-Type')
-    next();
+    next()
 }
 
 app.use(addCrossDomainHeaders)
 app.use(urlencodedParser)
 app.use(jsonParser)
 
+
+const botFactory = require('./bot')
```

Comment:
Please move the `require` to the top of the file... Also, I'd probably rename the required file to `bot-factory.js` now that it's a factory.